templates/endpoints_v2/CustomModels.md

```diff
@@ -0,0 +1,139 @@
+    downscale_delay_s: 300.0
+    upscale_delay_s: 90.0
+  ray_actor_options:
+    # Resources assigned to each model deployment. The deployment will be
+    # initialized first, and then start prediction workers which actually hold the model.
+    resources:
+      accelerator_type_cpu: 0.01
+engine_config:
+  # Model id - this is a RayLLM id
+  model_id: mosaicml/mpt-7b-instruct
+  # Id of the model on Hugging Face Hub. Defaults to model_id if not specified.
+  hf_model_id: mosaicml/mpt-7b-instruct
+  # vLLM keyword arguments passed when constructing the model.
+  engine_kwargs:
+    trust_remote_code: true
+  # Optional Ray Runtime Environment configuration. See Ray documentation for more details.
+  # Add dependent libraries, environment variables, etc.
+  runtime_env:
+    env_vars:
+      YOUR_ENV_VAR: "your_value"
+  # Optional configuration for loading the model from S3 instead of Hugging Face Hub. You can use this to speed up downloads or load models not on Hugging Face Hub.
+  s3_mirror_config:
+    bucket_uri: s3://large-dl-models-mirror/models--mosaicml--mpt-7b-instruct/main-safetensors/
+  generation:
+    # Prompt format to wrap queries in. {instruction} refers to user-supplied input.
+    prompt_format:
+      system: "{instruction}\n"  # System message. Will default to default_system_message
+      assistant: "### Response:\n{instruction}\n"  # Past assistant message. Used in chat completions API.
+      trailing_assistant: "### Response:\n"  # New assistant message. After this point, model will generate tokens.
+      user: "### Instruction:\n{instruction}\n"  # User message.
+      default_system_message: "Below is an instruction that describes a task. Write a response that appropriately completes the request."  # Default system message.
+      system_in_user: false  # Whether the system prompt is inside the user prompt. If true, the user field should include '{system}'
+      add_system_tags_even_if_message_is_empty: false  # Whether to include the system tags even if the user message is empty.
+      strip_whitespace: false  # Whether to automaticall strip whitespace from left and right of user supplied messages for chat completions
+    # Stopping sequences. The generation will stop when it encounters any of the sequences, or the tokenizer EOS token.
+    # Those can be strings, integers (token ids) or lists of integers.
+    # Stopping sequences supplied by the user in a request will be appended to this.
+    stopping_sequences: ["### Response:", "### End"]
+
+# Resources assigned to each model replica. This corresponds to Ray AIR ScalingConfig.
+scaling_config:
+  # If using multiple GPUs set num_gpus_per_worker to be 1 and then set num_workers to be the number of GPUs you want to use.
+  num_workers: 1
+  num_gpus_per_worker: 1
+  num_cpus_per_worker: 4
+  resources_per_worker:
+    # You can use custom resources to specify the instance type / accelerator type
+    # to use for the model.
+    accelerator_type_a10: 0.01
```

Comment:
list of allowed values may not be obvious to new users. We should probably indicate taht this should match custom resource definition in aws.yaml or gcp.yaml. Example `accelerator_type:A10G`

Reply:
We should stop using custom resources and support the proper way to request resources like (gpu:2, accelerator: A10G)

Reply:
i included it in OptimizeModels.md @gvspraveen , any other readme you think I should add it to? 
@scottsun94 placement groups don't support this resource right now

Reply:
> placement groups don't support this resource right now

Yeah. I meant that we should fix it down the road